net/ampr-ripd/files/ampr-ripd-init

```diff
@@ -0,0 +1,77 @@
+#!/bin/sh /etc/rc.common
+
+START=95
+STOP=10
+
+EXTRA_COMMANDS="configure"
+EXTRA_HELP="        configure       Configure service parameters"
+
+start() {
+   default_addr="44.127.254.254"
+   if [ $(uci get network.amprlan.ipaddr) = "$default_addr" ] | \
+      [ $(uci get network.amprwan.ipaddr) = "$default_addr" ]; then
+
+	cat <<-EOF
+	ampr-ripd is not fully configured.
+	You must run /etc/init.d/ampr-ripd configure.
+	EOF
+
+      exit 1
+   fi
+   if [ ! -d /var/lib/ampr-ripd ]; then
+      mkdir -p /var/lib/ampr-ripd
+   fi
+   ip tunnel change ttl 64 mode ipip tunl0
+   ip link set dev tunl0 up
+   ifconfig tunl0 mtu 1480
+   TUNNET=$(uci -q get network.amprlan.tunnet)
+   /usr/sbin/ampr-ripd -s -r -t 44 -i tunl0 -a $TUNNET
+}
+
+stop() {
+   ifconfig tunl0 down
+   killall ampr-ripd
+}
+
+configure() {
+   if [ -z $1 ] | [ -z $2 ] | [ -z $3 ]; then
+      echo Usage: /etc/init.d/ampr-ripd configure [amprhost] [amprmask] [amprnet]
+      exit 1
+   fi
+   amprhost=$1
+   amprmask=$2
+   amprnet=$3
+
+	cat <<-EOF
+	Configuring ampr-ripd with values:
+
+	amprhost=$amprhost
+	amprmask=$amprmask
+	amprnet=$amprnet
+	EOF
+
+   uci set network.amprlan.ipaddr=$amprhost
+   uci set network.amprlan.netmask=$amprmask
+   uci set network.amprlan.tunnet=$amprnet/$amprmask
```

Comment:
`tennet` doesn't seem like a standard network field. Consider a separate UCI-based config file. I've been suggesting this for a while now.

Reply:
Still unclear.